src/synapticon_interface.cpp

```diff
@@ -270,6 +270,8 @@ SynapticonSystemInterface::prepare_command_mode_switch(
         // Spring adjust puts all joints in QUICK_STOP mode except the spring adjust joint
         if (i == SPRING_ADJUST_JOINT_IDX) {
           new_modes.push_back(control_level_t::SPRING_ADJUST);
+          time_prev_ = std::chrono::steady_clock::now();
```

Comment:
Related to your comment about resetting `error_prev_` as well:

```suggestion
          error_prev_ = std::nullopt;
```